### PR TITLE
[WIP] 117 generator

### LINKED ISSUE #117: Add one step generator for new projects
Add:

## Example:
```bash
elmchemy new my_project
mix test 
# 0
```
->
```elixir
my_project/
├── elm-package.json
├── mix.exs
├── elm
    ├── Hello.elm
├── lib
    ├── Hello.ex
├── tests
    ├── hello_test.exs
```

### PR DESCRIPTION
**Pull request by Hajto**
One step generator based on the one that phoenix have.

[WIP] - Please don't touch since a lot of things will chagne

/Edit wende: Closes #117

### CODE REVIEW

**Comment by wende** [comment]
I'd appreciate some tests checking if generation works properly

File: installer/.gitignore [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+# The config/prod.secret.exs file by default contains sensitive
+# data and you should not commit it into version control.
+#
+# Alternatively, you may comment the line below and commit the
+# secrets file as long as you replace its contents by environment
+# variables.
+/config/prod.secret.exs
+
+# -*- mode: gitignore; -*-
```
Comment (wende) [inline installer/.gitignore]: Something went wrong here

Comment (Hajto) [inline installer/.gitignore]: What exactly?

Comment (wende) [inline installer/.gitignore]: As far as I know Emacs mode should be at the first line to work properly

**Comment by Baransu** [comment]
What the status of this PR. I would like to take it and lead it into merge.

/ cc @wende @Hajto

**Comment by Hajto** [comment]
@Baransu Go ahead, I am dead out of time to work on it. I will be gratefull!

File: installer/templates/new/elm-package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "summary": "helpful summary of your project, less than 80 characters",
+    "repository": "https://github.com/user/project.git",
+    "license": "BSD3",
+    "source-directories": [
+        "./elm"
+    ],
+    "exposed-modules": [],
+    "dependencies": {
+        "elm-lang/core": "5.1.1 <= v < 6.0.0",
+        "wende/elmchemy-core": "1.0.0 <= v < 2.0.0"
```
[the remark this answers is not in the document]
Comment (Hajto) [reply installer/templates/new/elm-package.json]: Done

File: installer/templates/new/test/test_template.exs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+defmodule CharacterTest do
+  use ExUnit.Case
+  use Elmchemy
```
[the remark this answers is not in the document]
Comment (Hajto) [reply installer/templates/new/test/test_template.exs]: Done

File: installer/.gitignore (hunk 1 above)
Comment (Hajto) [reply installer/.gitignore]: :/ As far as I can remember its fresh elixir generated gitignor

**Comment by wende** [comment]
Closing the PR till somebody picks it up

**Pull request closed without merging**